client/src/components/Login/SignUpForm/index.js

```diff
@@ -59,11 +61,60 @@ const SignUpForm = ({ switchView }) => {
     }
   };
 
+  const validateUsername = () => {
+    let usernameState = {
+      error: false,
+      message: ""
+    };
+    if (!username) {
+      usernameState.error = true;
+      usernameState.message = "Username is required";
+    }
+    setValidationState({
+      ...validationState,
+      username: usernameState
+    });
+  };
+
```

Comment:
You'll want to add a validateEmail method, which will just check that the email isn't blank (similar to the username validation).